NEXT_CHANGELOG.md

```diff
@@ -158,6 +158,15 @@ When receiving requests with invalid content-type/accept header missmatch (e.g m
 
 By [@Meemaw](https://github.com/Meemaw) in https://github.com/apollographql/router/pull/2370
 
+### Add a metric for existing connections ([issue #2384](https://github.com/apollographql/router/issues/2384))
+
+New metrics are available to track the client connections:
+- `apollo_router_session_count` indicates the number of currently connected clients
+- `apollo_router_active_session_count` indicates the number of in flight GraphQL requests from connected clients.
```

Comment:
I know this gets into code changes as well, but can we use tiered name-spacing/thoughtfully-ordered-prefixes to make the usage of this feature easier in tools in which these metrics are consumed?  For example, autocompletion will more clearly give you the answer you need in Datadog after you type `apollo_router_ses` — giving you the option to get `total` or `active`.

```suggestion
- `apollo_router_session_count` indicates the number of currently connected clients
- `apollo_router_session_count_active` indicates the number of in flight GraphQL requests from connected clients.
```

Additionally, I'm curious if `apollo_router_session_count` could have some suffix on it? Maybe `_total`, yielding `apollo_router_session_count_total`?

As a separate point, in today's setup I don't understand (based on what I read here) the nuance/difference between "number of currently connected clients" and "number of in-flight GraphQL requests from connected clients".  Is this accounting for multiple queries on the same connection due to multiplexing?

Reply:
the renaming makes sense, I'll add that. The difference between the two metrics here could appear in multiplexing if at some point we support client side HTTP/2, but right now the main difference we'll see is that `apollo_router_session_count = apollo_router_active_session_count + idle sessions`. This helps in capacity planning, because some services might get a lot connected clients that do not send queries very often

Reply:
renamed in  c4be9aa3